src/libnl3/patch/0002-mpls-remove-nl_addr_valid.patch

```diff
@@ -0,0 +1,55 @@
+From c89d1a129f71d3d2f76e6cbadb11ef41d8941a73 Mon Sep 17 00:00:00 2001
+From: Ann Pokora <apokora@juniper.net>
+Date: Tue, 25 May 2021 18:10:04 -0700
+Subject: [PATCH 2/2] mpls remove nl_addr_valid
```

Comment:
why are we remove the validation?

Reply:
@lguohan nl_addr_valid() arguments are supposed to be a pointer to an ASCII string and an address family value (AF_MPLS, AF_INET, AF_INET6)... it then uses these to validate that the string format contains an address for the specified family.
The removed calls to nl_addr_valid() were instead passing in a pointer to a binary address and the length of the binary address... this is completely wrong.  For most cases, the length of the binary address would not match any of the values of supported address families, so nl_addr_valid() would return true.  But if the binary address was an MPLS label stack with 3 labels, then the length would match the value of AF_DECnet (12) and it attempts to parse the binary address as if it were an ASCII string... this fails and nl_addr_valid() returns false.

Reply:
can you put the above explaination into the this patch file?

Reply:
@lguohan explanation is added to patch file.